cli/command/context/create.go

```diff
@@ -47,10 +47,13 @@ func longCreateDescription() string {
 func newCreateCommand(dockerCli command.Cli) *cobra.Command {
 	opts := &CreateOptions{}
 	cmd := &cobra.Command{
-		Use:   "create [OPTIONS] CONTEXT",
+		Use:   "create [OPTIONS] [TYPE] CONTEXT",
 		Short: "Create a context",
-		Args:  cli.ExactArgs(1),
+		Args:  cli.RequiresRangeArgs(1, 2),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 2 {
```

Comment:
At this level, we'll have to find a way to accept ACI/ECS specific flags, I believe it will fail before reaching this if you run `docker context create aci my context --subscription-id xxx`. 
Maybe it will be easier to defined the create subcommands we support and simply exec from there ? (duplicating commands, but ... ). 
For now at least, we want a specified list of context types we can create, having this in `--help` will be nicer too

Reply:
(same question for login)

Reply:
oh yes indeed, this one will require a less-naive approach

Reply:
I've found a hack, but it's really a crappy hack :'(